rust/crd/src/security.rs

```diff
@@ -0,0 +1,504 @@
+    const STACKABLE_TLS_SERVER_DIR: &'static str = "/stackable/tls_server";
+    const STACKABLE_TLS_INTERNAL_MOUNT_DIR: &'static str = "/stackable/tls_internal_mount";
+    const STACKABLE_TLS_INTERNAL_DIR: &'static str = "/stackable/tls_internal";
+    const SYSTEM_TRUST_STORE_DIR: &'static str = "/etc/pki/java/cacerts";
+
+    pub fn new(
+        resolved_authentication_classes: ResolvedAuthenticationClasses,
+        internal_secret_class: String,
+        server_secret_class: Option<String>,
+    ) -> Self {
+        Self {
+            resolved_authentication_classes,
+            internal_secret_class,
+            server_secret_class,
+        }
+    }
+
+    /// Create a `KafkaSecurity` struct from the Kafka custom resource and resolve
+    /// all provided `AuthenticationClass` references.
+    pub async fn new_from_kafka_cluster(
+        client: &Client,
+        kafka: &KafkaCluster,
+    ) -> Result<Self, Error> {
+        Ok(KafkaSecurity {
+            resolved_authentication_classes: authentication::resolve_authentication_classes(
+                client,
+                &kafka.spec.cluster_config.authentication,
+            )
+            .await
+            .context(InvalidAuthenticationClassConfigurationSnafu)?,
+            internal_secret_class: kafka
+                .spec
+                .cluster_config
+                .tls
+                .as_ref()
+                .map(|tls| tls.internal_secret_class.clone())
+                .unwrap_or_else(tls::internal_tls_default),
+            server_secret_class: kafka
+                .spec
+                .cluster_config
+                .tls
+                .as_ref()
+                .and_then(|tls| tls.server_secret_class.clone()),
+        })
+    }
+
+    /// Check if TLS encryption is enabled. This could be due to:
+    /// - A provided server `SecretClass`
+    /// - A provided client `AuthenticationClass`
+    /// This affects init container commands, Kafka configuration, volume mounts and
+    /// the Kafka client port
+    pub fn tls_enabled(&self) -> bool {
+        // TODO: This must be adapted if other authentication methods are supported and require TLS
+        self.tls_client_authentication_class().is_some() || self.tls_server_secret_class().is_some()
+    }
+
+    /// Retrieve an optional TLS secret class for external client -> server communications.
+    pub fn tls_server_secret_class(&self) -> Option<&str> {
+        self.server_secret_class.as_deref()
+    }
+
+    /// Retrieve an optional TLS `AuthenticationClass`.
+    pub fn tls_client_authentication_class(&self) -> Option<&AuthenticationClass> {
+        self.resolved_authentication_classes
+            .get_tls_authentication_class()
+    }
+
+    /// Retrieve the mandatory internal `SecretClass`.
+    pub fn tls_internal_secret_class(&self) -> Option<&str> {
+        // TODO: which check here - this can/should not be happening?
```

Comment:
Anything you want to do here before merging?

Reply:
I removed the TODO, nothing we can do there :D